### PR TITLE
media state: replace `MediaActions.edit` with redux actions/thunks

### LINKED ISSUE #44001: Media: Replace usages of flux edit action with redux action
Update areas calling to `MediaActions.edit` or `update` to call the redux action

- [x] `edit`
- [x] `update`

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

Note: `MediaActions.edit` is a cosmetic call which only updates the local store with no connection to update data on the server. I went for more specific functions (e.g. `clearMediaItemStatus`) as I wasn't happy with creating a general `"cosmeticallyEditMediaItemLocally"` media thunk.

* Remove `MediaActions.edit` method from `lib/media/actions`
* VideoEditor: move `updatePoster` functionality to `onSuccess` handler and get rid of `MediaActions.edit` call
* Post Editor: Replace `MediaActions.edit` call with `clearMediaItemDirtyStatus` thunk

##### Video Editor

1. Go to `/media` on a site which contains video items
2. Update the video thumbnail and make sure you see the changes after saving

##### Post Editor

1. Open up the classic post editor on a post which contains at least two copies of the same image
2. Open the media item details dialog and edit one of the items (e.g. flip vertically), hit save
3. Now make sure all the items in the editor get updated accordingly
4. Inspect the flux media store and make sure the corresponding media item dirty status is set to `isDirty: false`

fixes #44001

### CODE REVIEW

File: client/components/tinymce/plugins/media/plugin.jsx [hunk 1]
```diff
@@ -387,7 +386,7 @@ function mediaButton( editor ) {
 		// After editing an image, we need to update them in a post/page. If we updated all instances
 		// of the edited image, we dispatch an action which marks the image media object not dirty.
 		if ( lastDirtyImage && lastDirtyImage.isDirty && numOfImagesToUpdate === 0 ) {
-			MediaActions.edit( selectedSite.ID, { ...lastDirtyImage, isDirty: false } );
+			dispatch( clearMediaItemDirtyStatus( selectedSite.ID, lastDirtyImage.ID ) );
```
Comment (saramarcondes) [inline client/components/tinymce/plugins/media/plugin.jsx:389]: I love this refactor!

**Comment by flootr** [comment]
> I edited the second one of the image but only the first one in the post actually reflected the change. You can see that both are the same image.

Ok, this one took me quite a while and I'm still not sure I understand it fully. I was able to reproduce that not all images get updated after I edit the source. However, I could only reproduce it the first time this image got inserted into the post. All subsequent edits worked correctly 🤯 after I tried a few different things (like moving `isDirty` completely over to redux) which all didn't work, I completely removed the dirty status clearance (ec100d5). Now that seems to work well 🙈 would you mind trying that on your end @saramarcondes ?

My guess why this still works after removing the code is that after we do the edit request to the server the item gets replaced in the store and the `isDirty` flag is removed as part of this flow. We still need to track `isDirty` so all image representations get updated correctly though.

Looping @lamosty in here as it seems you wrote that code originally and might have more insight on what's going on :) although it's been a while.

**Comment by saramarcondes** [comment]
@flootr Wow, how strange! Having still only a vague understanding of this area, your explanation makes sense to me. We do indeed simply replace the media in the store with the new version.

We may want to update/reword the comments around where you removed the dispatch to the clean dirty status thunk. Right now it might be surprising that we don't explicitly do it.

Potentially we would also want to add a comment to the `media/reducer.transientItems` in the `MEDIA_RECEIVE` case as I believe that's where we're replacing the old media with the new non-dirty one, right?

**Comment by lamosty** [comment]
> Looping @lamosty in here as it seems you wrote that code originally and might have more insight on what's going on :) although it's been a while.

Hey, thanks for the ping.

Yeah, it's pretty old code to me and apart from the comment, I don't remember the exact details; if you still need help I can study and dig through the code like it was not originally written by me :P

**Pull request merged**